File: front/src/store/user_event.js
import instance from "@/middlewares";


export default {
  name: "user_event",
  state: () => ({
    data: null,

  }),
  mutations: {
    setData(state, data) {
      state.data = data;
    },


  },
  actions: {

    // get user events for user uid
    async getUserEvents({ commit }, user_uid) {
      try {
        const response = await instance.get(`/api/user_event/user/${user_uid}`);
        if (response) return commit("setData", response.data);
      } catch (error) {
        console.log(error);
      }
    },
    // get user events for event id
    async getUserEventsByEventId({ commit }, event_id) {
      try {
        const response = await instance.get(`/api/user_event/event/${event_id}`);
        if (response) return commit("setData", response.data);
      } catch (error) {
        console.log(error);
      }
    },
    // create a new user event
    async createUserEvent({}, input) {
      try {
        const { user_uid, event_id } = input;
        const response = await instance.post("/api/user_event", { user_uid, event_id });
        if (response.ok) return console.log("ok");
      } catch (error) {
        console.log(error);
      }
    },
    // update an existing user event by id
    async updateUserEvent({}, input) {
      try {
        const { user_event_id, user_uid, event_id } = input;
        const response = await instance.put(`/api/user_event/${user_event_id}`, { user_uid, event_id });
        if (response.ok) return console.log("ok");
      } catch (error) {
        console.log(error);
      }
    },
    // delete an existing user event by id
    async deleteUserEvent({}, user_event_id) {
      try {
        const response = await instance.delete(`/api/user_event/${user_event_id}`);
        if (response.ok) return console.log("ok");
      } catch (error) {
        console.log(error);
      }
    },


},

namespaced: true,
};

File: backend/app/routers/user_event.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.schemas.user_event import UserEventCreate, UserEventUpdate, UserEventOut
from app.controllers.user_event import create_user_event, get_user_event_by_uid, get_user_event_by_event_id, update_user_event, delete_user_event

from app.database import get_db

router = APIRouter()

# get user events for user uid
@router.get("/user/{user_uid}", response_model=list[UserEventOut])
def get_user_events(user_uid: str, db: Session = Depends(get_db)):
    return get_user_event_by_uid(db, user_uid)


# get user events for event id
@router.get("/event/{event_id}", response_model=list[UserEventOut])
def get_user_events(event_id: int, db: Session = Depends(get_db)):
    return get_user_event_by_event_id(db, event_id)

# create a new user event
@router.post("/", response_model=UserEventOut)
def create_user_event_route(user_event: UserEventCreate, db: Session = Depends(get_db)):
    return create_user_event(db, user_event)


# update an existing user event by id
@router.put("/{user_event_id}", response_model=UserEventOut)
def update_user_event_route(user_event_id: int, user_event: UserEventUpdate, db: Session = Depends(get_db)):
    return update_user_event(db, user_event_id, user_event)


# delete an existing user event by id
@router.delete("/{user_event_id}", response_model=UserEventOut)
def delete_user_event_route(user_event_id: int, db: Session = Depends(get_db)):
    return delete_user_event(db, user_event_id)